src/Model/Rule/Condition/ShopgateOrder.php

```diff
@@ -0,0 +1,137 @@
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+namespace Shopgate\Base\Model\Rule\Condition;
+
+use Magento\Rule\Model\Condition\AbstractCondition;
+use ShopgateClient;
+
+class ShopgateOrder extends AbstractCondition
+{
+    const CLIENT_ATTRIBUTE = 'shopgate_client';
+    const APP_CLIENTS      = [
+        ShopgateClient::TYPE_IPHONEAPP,
+        ShopgateClient::TYPE_IPADAPP,
+        ShopgateClient::TYPE_ANDROIDPHONEAPP,
+        ShopgateClient::TYPE_ANDROIDTABLETAPP,
+    ];
+
+    /**
+     * @var \Magento\Config\Model\Config\Source\Yesno
+     */
+    protected $sourceYesno;
+
+    /**
+     * @var \Magento\Sales\Model\ResourceModel\Order\CollectionFactory
+     */
+    protected $orderFactory;
+
+    /**
+     * Constructor
+     *
+     * @param \Magento\Rule\Model\Condition\Context                      $context
+     * @param \Magento\Config\Model\Config\Source\Yesno                  $sourceYesno
+     * @param \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderFactory
+     * @param array                                                      $data
+     */
+    public function __construct(
+        \Magento\Rule\Model\Condition\Context $context,
+        \Magento\Config\Model\Config\Source\Yesno $sourceYesno,
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderFactory,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->sourceYesno  = $sourceYesno;
+        $this->orderFactory = $orderFactory;
+    }
+
+    /**
+     * Load attribute options
+     *
+     * @return $this
+     */
+    public function loadAttributeOptions()
+    {
+        $this->setAttributeOption([
+            'is_shopgate_order' => __('is mobile app'),
+        ]
+        );
+
+        return $this;
+    }
+
+    /**
+     * Get input type
+     *
+     * @return string
+     */
+    public function getInputType()
+    {
+        return 'select';
+    }
+
+    /**
+     * Get value element type
+     *
+     * @return string
+     */
+    public function getValueElementType()
+    {
+        return 'select';
+    }
+
+    /**
+     * Get value select options
+     *
+     * @return array|mixed
+     */
+    public function getValueSelectOptions()
+    {
+        if (!$this->hasData('value_select_options')) {
+            $this->setData(
+                'value_select_options',
+                $this->sourceYesno->toOptionArray()
+            );
+        }
+
+        return $this->getData('value_select_options');
+    }
+
+    /**
+     * Validate App Rule Condition
+     *
+     * @param \Magento\Framework\Model\AbstractModel $model
+     *
+     * @return bool
+     */
+    public function validate(\Magento\Framework\Model\AbstractModel $model)
+    {
+        $isShopgateOrder = 0;
+        if (in_array($model->getData(self::CLIENT_ATTRIBUTE), self::APP_CLIENTS)) {
```

Comment:
`$isSgOrder = (int) in_array(...);` soooo pretty!

Reply:
the whole point is, that later we can add
        if (is_app_webcheckout) {
            $isShopgateOrder = 1;
        } 
but I guess we should not prepare code that we won't add for at least a few months.
Done